src/context-selection/org-unit-selector-bar-item/use-load-offline-levels.js

```diff
@@ -0,0 +1,43 @@
+import { useDataEngine } from '@dhis2/app-runtime'
+import { useEffect, useState } from 'react'
+import loadOfflineLevel from './load-offline-level.js'
+import useOfflineLevelsToLoad from './use-offline-levels-to-load.js'
+import useOrganisationUnitLevels from './use-organisation-unit-levels.js'
+
+/**
+ * As the service worker caches request responses,
+ * it should suffice to simply perform all request the org unit tree would
+ * perform as well in advance
+ */
```

Comment:
Since we're already prefetching content with react query (the custom forms) I think it would also make sense to use react-query to prefetch here as well. That means the SW can be confined to just simple caching and means we aren't mixing cache strategies across the app.